File: 2024/august/javaScript/05.js
/**
 * @param {string[]} arr
 * @param {number} k
 * @return {string}
 */
var kthDistinct = function(arr, k) {
        let f = new Map();
        for (const ch of arr){
            if (f.has(ch)){
                f.set(ch, f.get(ch)+1);
            }else {
                f.set(ch , 1);
            }

        }
        for (const cur of arr){
            let cnt = f.get(cur);
            if (cnt == 1) --k;
            if (k == 0) return cur;
        }
        return "";
};

File: 2024/august/java/05.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        var f = new HashMap<String , Integer> ();
        for (var cur : arr){
            f.put(cur , f.getOrDefault(cur , 0)+1);
        }
        for (var cur : arr){
            int cnt = f.get(cur);
            if (cnt == 1) --k;
            if (k == 0) return cur;
        }
        assert(k > 0);
        return "";
    }
}
